mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+    _unpack_and_save_model(spark_model, tmp_path)
+    sparkml_data_path = os.path.abspath(os.path.join(path, _JOHNSNOWLABS_MODEL_PATH_SUB))
+    # We're copying the Spark model from DBFS to the local filesystem if (a) the temporary DFS URI
+    # we saved the Spark model to is a DBFS URI ("dbfs:/my-directory"), or (b) if we're running
+    # on a Databricks cluster and the URI is schemeless (e.g. looks like a filesystem absolute path
+    # like "/my-directory")
+    copying_from_dbfs = is_valid_dbfs_uri(tmp_path) or (
+            databricks_utils.is_in_cluster() and posixpath.abspath(tmp_path) == tmp_path
+    )
+    if copying_from_dbfs and databricks_utils.is_dbfs_fuse_available():
+        tmp_path_fuse = dbfs_hdfs_uri_to_fuse_path(tmp_path)
+        shutil.move(src=tmp_path_fuse, dst=sparkml_data_path)
+    else:
+        _HadoopFileSystem.copy_to_local_file(tmp_path, sparkml_data_path, remove_src=True)
+    _save_model_metadata(
+        dst_dir=path,
+        spark_model=spark_model,
+        mlflow_model=mlflow_model,
+        sample_input=sample_input,
+        conda_env=conda_env,
+        code_paths=code_paths,
+        signature=signature,
+        input_example=input_example,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+    )
+
+
+def _load_model_databricks(dfs_tmpdir, local_model_path):
+    # Spark ML expects the model to be stored on DFS
+    # Copy the model to a temp DFS location first. We cannot delete this file, as
+    # Spark may read from it at any point.
+    fuse_dfs_tmpdir = dbfs_hdfs_uri_to_fuse_path(dfs_tmpdir)
+    os.makedirs(fuse_dfs_tmpdir)
+    # Workaround for inability to use shutil.copytree with DBFS FUSE due to permission-denied
+    # errors on passthrough-enabled clusters when attempting to copy permission bits for directories
+    shutil_copytree_without_file_permissions(src_dir=local_model_path, dst_dir=fuse_dfs_tmpdir)
+    return nlp.load(path=dfs_tmpdir)
+
+
+def _load_model(model_uri, dfs_tmpdir_base=None, local_model_path=None):
+    dfs_tmpdir = generate_tmp_dfs_path(dfs_tmpdir_base or MLFLOW_DFS_TMP.get())
+    if databricks_utils.is_in_cluster() and databricks_utils.is_dbfs_fuse_available():
+        return _load_model_databricks(
+            dfs_tmpdir, local_model_path or _download_artifact_from_uri(model_uri)
+        )
+    # model_uri = _HadoopFileSystem.maybe_copy_from_uri(model_uri, dfs_tmpdir, local_model_path)
+    if model_uri and not local_model_path:
+        local_model_path = _download_artifact_from_uri(model_uri)
+    get_or_create_sparksession(local_model_path)
+
+    if _JOHNSNOWLABS_MODEL_PATH_SUB not in local_model_path:
+        local_model_path = os.path.join(local_model_path, _JOHNSNOWLABS_MODEL_PATH_SUB)
+
+    return nlp.load(path=local_model_path)
+
+
+def load_model(model_uri, dfs_tmpdir=None, dst_path=None, **kwargs):
+    """
+    Load the Spark MLlib model from the path.
```

Comment:
Please update this comment with flavor-specific information and references

Reply:
thanks updated